cmd/log-validator/main.go

```diff
@@ -43,9 +44,16 @@ func lineValid(text string) error {
 	checksum := fields[5]
 	// Reconstruct just the message portion of the line
 	line := strings.Join(fields[6:], " ")
+
+	const errorPrefix = "invalid checksum"
+	// If we are fed our own output, treat it as always valid. This
+	// prevents runaway scenarios where we generate ever-longer output.
+	if strings.Contains(text, errorPrefix) {
```

Comment:
Can this not be `strings.HasPrefix(line, errorPrefix)` now?

Reply:
Yes, in theory; but the issue that caused us to generate excessive log lines was that the input was not formatted as expected. If that issue recurs, the errorPrefix could occur at some arbitrary part of the `text`, either before or after the beginning of the `line` proper.